apigentools/commands/generate.py

```diff
@@ -580,3 +593,40 @@ def pull_repository(self, language, branch=None):
                         f"Could not merge {self.args.get('is_ancestor')} to {branch} to keep it up-to-date"
                     )
                     raise
+
+    def remove_generated_files(self, language):
+        """
+        Remove all generated files from the generate output directory
+        Files are deemed as "generated" if they match any regex in the .generated_files file
+        at the root of the output repository.
+        """
+        blacklist_regexes = set()
+        output_dir = os.path.abspath(self.get_generated_lang_dir(language))
+        blacklist_file = os.path.join(output_dir, GENERATION_BLACKLIST_FILENAME)
+
+        log.info(f"Removing generated files from the output directory: {output_dir}")
+
+        if not os.path.exists(blacklist_file):
+            log.warning(
+                f"File: {blacklist_file} doesn't exist, skipping removal of generated files"
+            )
+            return
+
+        # We should already be in this directory, but its explicit and safer since we're deleting
+        with change_cwd(output_dir):
+            # Read in and compile the regexes of files we want to delete
+            with open(blacklist_file, "r") as blacklist_file:
+                for line in blacklist_file.readlines():
+                    blacklist_regexes.add(re.compile(line.strip()))
+
+            # Get all files from current directory recursively
+            all_files = [
+                os.path.relpath(os.path.join(root, filename), start=output_dir)
+                for root, _, files in os.walk(output_dir)
+                for filename in files
+            ]
+            # Match the regex against the list of all files and delete
+            for file in all_files:
+                if any(blacklist_regex.match(file) for blacklist_regex in blacklist_regexes):
+                        log.debug(f"Removing generated file: {file}")
+                        os.remove(file)
```

Comment:
Oups forgot indent in my suggestion 🙈 
```suggestion
                    log.debug(f"Removing generated file: {file}")
                    os.remove(file)
```